File: backend/app/models/source.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Text, func

from app.database import Base


class Source(Base):
    __tablename__ = "sources"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(255), nullable=False)
    source_type = Column(String(20), nullable=False)  # document, link, image, audio, video
    file_type = Column(String(20), nullable=False)
    file_size = Column(Integer, nullable=True)
    file_path = Column(Text, nullable=True)
    url = Column(Text, nullable=True)
    content_text = Column(Text, nullable=True)  # Extracted plaintext for RAG
    content_hash = Column(String(64), nullable=True)  # SHA256 for extraction cache
    status = Column(String(20), nullable=False, default="uploaded")
    created_at = Column(DateTime, server_default=func.now())

File: frontend/src/types/source.ts
export type SourceType = 'document' | 'link' | 'image' | 'audio' | 'video'
export type ParseStatus = 'uploaded' | 'fetching' | 'parsing' | 'extracting' | 'parsed' | 'failed' | 'unsupported'

export interface Source {
  id: number
  name: string
  source_type: SourceType
  file_type: string
  file_size: number | null
  url: string | null
  status: ParseStatus
  created_at: string
}
